main.go

```diff
@@ -74,17 +76,29 @@ func main() {
 		PublishInternal:  cfg.PublishInternal,
 	}
 
-	// Lookup all the selected sources by names and pass them the desired configuration.
-	sources, err := source.ByNames(&source.SingletonClientGenerator{
-		KubeConfig: cfg.KubeConfig,
-		KubeMaster: cfg.Master,
-	}, cfg.Sources, sourceCfg)
-	if err != nil {
-		log.Fatal(err)
-	}
+	var endpointsSource source.Source
+
+	if cfg.Cleanup {
```

Comment:
Could we instead solve this by forcing ExternalDNS to not find any desired Endpoint? For instance, when deleting a DaemonSet Kubernetes will first update the DaemonSet with a non-matching nodeSelector (e.g. `someUID=someUID`) so that the Pods get unscheduled and terminated from the nodes. Once everything is gone it deleted the DaemonSet itself.

`--cleanup` could just be a shortcut for such a non-matching selector internally which would avoid the special case here.

Reply:
Basically the proposed solution just assures that no endpont is found. It omits the regular endpoint sources, so no endpoint will be found and the regular logic just deletes all currently active DNS entries. No explicit coding to delete the entries has been added. This seems to be nearly your proposal.

So far at least one endpoint source is required, so it is not possible to call ExternalDNS without any endpoint source,
This could also be changed, so not specifying an endpoint source would implicitly force the tool to delete all existing DNS entries. Nevertheless I think,  it should be a better solution to explicitly tell the tool to cleanup all entries instead of an implicit behaviour.

If the cleanup option just adds a new *fake* ingress source threre would be the need for extra coding preventing to add additional regular sources, because the basic behaviour is accumulative. And combining such an empty endpoint source with regular sources would be quite confusing. In the proposed solution the extra coding required for the cleanup option is just a single if statement.

But if you have an easier more obvious solution, no problem. All I wanted was to have a possibility to cleanup all the generated DNS entries without manually identifying and deleting the entries on the IaaS layers,